src/app/donation-start/donation-start.component.ts

```diff
@@ -729,13 +729,32 @@ export class DonationStartComponent implements AfterContentChecked, AfterContent
   }
 
   captchaDonationReturn(captchaResponse: string) {
+    if (captchaResponse === null) {
+      // Ensure no other callback tries to use the old captcha code, and will re-execute
+      // the catcha to get a new one as needed instead.
+      this.captchaCode = undefined;
+      return;
+    }
+
     this.captchaCode = captchaResponse;
-    this.createDonationAndMaybePerson();
+
+    if (!this.donation) {
+      this.createDonationAndMaybePerson();
```

Comment:
`this.createDonationAndMaybePerson()` is called from 4 different places in the same file, but you've only added the 'falsey' check for the two captchaReturn functions. Shouldn't the two other calls of said function also have the `if` check. And if so, shouldn't we just put the `if` check as the first thing inside the `createDonationAndMaybePerson()` function definition?

Reply:
@alihejazi I did think about this but as we have identified the specific cause of issues as the captcha resolve callback, I would rather change only that fn's behaviour for now. We don't know without a much more time consuming audit if there could be edge cases that tacitly rely on an existing `this.donation` being updated in-place.